mm2src/lp_ordermatch.rs

```diff
@@ -2825,50 +2818,53 @@ pub async fn lp_ordermatch_loop(ctx: MmArc) {
         }
 
         {
-            let my_maker_orders = ordermatch_ctx.my_maker_orders.lock().clone();
+            let mut missing_uuids = Vec::new();
+            {
+                let orderbook = ordermatch_ctx.orderbook.lock();
```

Comment:
What do you think if we avoid locking two mutexes by cloning order uuids?
```rust
let known_uuids: HashSet<Uuid> = ordermatch.ctx.orderbook.lock().order_set.iter().map(|(uuid, _)| *uuid).collect();
```

Reply:
This is what I actually tried to avoid: the problem of "leak" is the desync between `my_maker_orders` and `orderbook` states: while we operate with a cloned version of the former, some orders can be canceled and removed from the latter. But we will re-add them again. I'm even thinking about having a new struct like
```rust
struct MyOrderbookState {
    my_maker_orders: HashMap<...>,
    my_pubkey_state: OrderbookPubkeyState
}
```
that will help us to always perform synchronized changes and avoid race conditions.

Reply:
Got it, thanks!